06.bowling_object/game.rb

```diff
@@ -0,0 +1,99 @@
+# frozen_string_literal: true
+
+require_relative 'shot'
+require_relative 'frame'
+
+class Game
+  def initialize(mark)
+    @mark = mark
+    make_shots
+    @frames = []
+    @count_frames = []
+  end
+
+  def make_shots
+    marks = @mark.split(',')
+
+    @shots = marks.map do |mark|
+      Shot.new(mark).score
+    end
+  end
+
+  def make_frames
+    @shots.each_with_index do |shot, index|
+      count_shot(shot, index)
+    end
+    @frames.map(&:make_frame)
+  end
+
+  def count_shot(shot, index)
+    @count_frames << shot
+    if @frames.size < 9
+      regular_frame(shot)
+    else
+      last_frame(index)
+    end
+  end
+
+  def regular_frame(shot)
+    if @count_frames.size == 2
+      @frames << Frame.new(@count_frames[0], @count_frames[1])
+      @count_frames.clear
+    elsif shot == 10
+      @frames << Frame.new(shot, nil)
+      @count_frames.clear
+    end
+  end
+
+  def last_frame(index)
+    return unless @frames.size == 9
+
+    remaining_shots = @shots[index..]
+    @frames << if remaining_shots.size == 2
+                 Frame.new(remaining_shots[0], remaining_shots[1])
+               else
+                 Frame.new(remaining_shots[0], remaining_shots[1], remaining_shots[2])
+               end
```

Comment:
```suggestion
    @frames << Frame.new(*remaining_shots[0..2])
```
でもよいかなと思いました。
https://docs.ruby-lang.org/ja/latest/method/Array/i/=5b=5d.html

Reply:
こんなのあるんですね🙌分岐がなくなりすごいスッキリします。